src/core/mac/mac.cpp

```diff
@@ -569,6 +596,12 @@ void Mac::ScheduleNextTransmission(void)
     else
     {
         mState = kStateIdle;
+
+        if (mDisableRadioOnIdle)
+        {
+            mDisableRadioOnIdle = false;
+            DisableRadio();
+        }
```

Comment:
I think we do need to support these scenarios...
I can easily envision multiple situations where this sequence of commands would happen.